Last login: Tue Jan 11 20:08:38 on ttys000
!%                                                                              (base) Angela@gwagsalaui-MacBook-Pro ~ % !pip install rpy2
pip install plotly --upgrade install -U finance-datareader install rpy2
Requirement already satisfied: plotly in ./opt/anaconda3/lib/python3.9/site-packages (5.4.0)
Collecting plotly
  Downloading plotly-5.5.0-py2.py3-none-any.whl (26.5 MB)
     |████████████████████████████████| 26.5 MB 739 kB/s 
Requirement already satisfied: install in ./opt/anaconda3/lib/python3.9/site-packages (1.3.5)
Requirement already satisfied: finance-datareader in ./opt/anaconda3/lib/python3.9/site-packages (0.9.31)
Collecting rpy2
  Downloading rpy2-3.4.5.tar.gz (194 kB)
     |████████████████████████████████| 194 kB 23.3 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_fae97696f00f4f7a96f9844bee44ab91/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_fae97696f00f4f7a96f9844bee44ab91/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-h_17u4vn
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_fae97696f00f4f7a96f9844bee44ab91/
    Complete output (2 lines):
    Error: rpy2 in API mode cannot be built without R in the PATH or R_HOME defined. Correct this or force ABI mode-only by defining the environment variable RPY2_CFFI_MODE=ABI
    ['cffi>=1.10.0', 'jinja2', 'pytz', 'tzlocal']
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/6d/c0/cc5fc7704f50e3af89fc4381afcb1faaa691cf0ec8a452fcbdc41cf85fc8/rpy2-3.4.5.tar.gz#sha256=5d31a5ea43f5a59f6dec30faca87edb01fc9b8affa0beae96a99be923bd7dab3 (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.4.4.tar.gz (187 kB)
     |████████████████████████████████| 187 kB 31.3 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_5e342c91a42642238df9a44f6a98c0df/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_5e342c91a42642238df9a44f6a98c0df/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-6gxs37ft
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_5e342c91a42642238df9a44f6a98c0df/
    Complete output (2 lines):
    Error: rpy2 in API mode cannot be built without R in the PATH or R_HOME defined. Correct this or force ABI mode-only by defining the environment variable RPY2_CFFI_MODE=ABI
    ['cffi>=1.10.0', 'jinja2', 'pytz', 'tzlocal']
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/cb/a5/354fdfc13ac30fd496c100a66587bbf7f0adcd320fa751311566f198bdbd/rpy2-3.4.4.tar.gz#sha256=e198862701bfc6d3da4a2fcf18910d71521b385283abdb2121e07b874add8e50 (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.4.3.tar.gz (188 kB)
     |████████████████████████████████| 188 kB 39.2 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_84f2b92fdf7f4801a0180c061ee5bc56/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_84f2b92fdf7f4801a0180c061ee5bc56/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-89xxsatx
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_84f2b92fdf7f4801a0180c061ee5bc56/
    Complete output (2 lines):
    Error: rpy2 in API mode cannot be built without R in the PATH or R_HOME defined. Correct this or force ABI mode-only by defining the environment variable RPY2_CFFI_MODE=ABI
    ['cffi>=1.10.0', 'jinja2', 'pytz', 'tzlocal']
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/38/d4/8e79a52dda0f41fced4381f2964c592573ac37df14480be75fbdd55e5587/rpy2-3.4.3.tar.gz#sha256=a39f2d75c24c688d5f48dfb2ef82efc006f2a51591941743026e1182353bf558 (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.4.2.tar.gz (186 kB)
     |████████████████████████████████| 186 kB 34.5 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_f88f935a38624734a1835a4f4127b3d8/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_f88f935a38624734a1835a4f4127b3d8/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-ib1s6hpw
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_f88f935a38624734a1835a4f4127b3d8/
    Complete output (2 lines):
    Error: rpy2 in API mode cannot be built without R in the PATH or R_HOME defined. Correct this or force ABI mode-only by defining the environment variable RPY2_CFFI_MODE=ABI
    ['cffi>=1.10.0', 'jinja2', 'pytz', 'tzlocal']
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/3c/27/035b298de7fab4ca97e46c12187b6eb487a3bd937a1e3512f28ef8ba8961/rpy2-3.4.2.tar.gz#sha256=8f7d1348b77bc45425b846a0d625f24a51a1c4f32ef2cd1c07a24222aa64e2e0 (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.4.1.tar.gz (185 kB)
     |████████████████████████████████| 185 kB 40.9 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_984d29ca42c642fda1d43b8326a60dd3/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_984d29ca42c642fda1d43b8326a60dd3/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-ckxc9fxq
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_984d29ca42c642fda1d43b8326a60dd3/
    Complete output (2 lines):
    Error: rpy2 in API mode cannot be built without R in the PATH or R_HOME defined. Correct this or force ABI mode-only by defining the environment variable RPY2_CFFI_MODE=ABI
    ['cffi>=1.10.0', 'jinja2', 'pytest', 'pytz', 'tzlocal']
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/4e/df/9a745e28de76e3905da3d30b31f10440b7e79390f611bdbaee54f1317efd/rpy2-3.4.1.tar.gz#sha256=644360b569656700dfe13f59878ec1cf8c116c128d4f2f0bf96144031f95d2e2 (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.4.0.tar.gz (185 kB)
     |████████████████████████████████| 185 kB 34.7 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_9d4bdf01674c453085ab9fdfc28577c2/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_9d4bdf01674c453085ab9fdfc28577c2/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-ap3obgiu
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_9d4bdf01674c453085ab9fdfc28577c2/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_9d4bdf01674c453085ab9fdfc28577c2/setup.py", line 182, in <module>
        with open(
    FileNotFoundError: [Errno 2] No such file or directory: '/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_9d4bdf01674c453085ab9fdfc28577c2/requirements.txt'
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/3c/b9/e9f114aee7cc1d86c1e916fee197e35243d115a26130bd4d4a41ac337421/rpy2-3.4.0.tar.gz#sha256=d543e4a577961a24a79cfa3e0c31cdc0169cdcb4b7d80d035f4bcc9df8102d80 (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.3.6.tar.gz (179 kB)
     |████████████████████████████████| 179 kB 41.4 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_6242027b4a1547c3aae5ac5a09375af2/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_6242027b4a1547c3aae5ac5a09375af2/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-ai_pvwsc
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_6242027b4a1547c3aae5ac5a09375af2/
    Complete output (2 lines):
    Error: rpy2 in API mode cannot be built without R in the PATH or R_HOME defined. Correct this or force ABI mode-only by defining the environment variable RPY2_CFFI_MODE=ABI
    cffi mode: CFFI_MODE.ANY
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/24/09/85edceaad76c9e99d93701b0d968593489ce66bc8c5ee58378774a8f8d5d/rpy2-3.3.6.tar.gz#sha256=ce063f3286e717b3914728ad23ec7db0a0f117ba3ade5ada8a250700779f6e77 (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.3.5.tar.gz (174 kB)
     |████████████████████████████████| 174 kB 30.5 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_071aef61be704940a3b25b9c5d4ff3d2/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_071aef61be704940a3b25b9c5d4ff3d2/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-p62pm_b5
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_071aef61be704940a3b25b9c5d4ff3d2/
    Complete output (2 lines):
    Error: rpy2 in API mode cannot be built without R in the PATH or R_HOME defined. Correct this or force ABI mode-only by defining the environment variable RPY2_CFFI_MODE=ABI
    cffi mode: CFFI_MODE.ANY
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/c5/d8/e63f98aa4a9912a19d5244cdf40a9908a4f5f99f84aa6338a0eb9b7ff048/rpy2-3.3.5.tar.gz#sha256=d2b1fb0e22924b08ba11acf07b93cec1842296c005e142834008bba25b3643af (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.3.4.tar.gz (174 kB)
     |████████████████████████████████| 174 kB 29.7 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_73c082c32cc64de981eb183848f6bcf7/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_73c082c32cc64de981eb183848f6bcf7/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-enbdp5kd
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_73c082c32cc64de981eb183848f6bcf7/
    Complete output (2 lines):
    Error: rpy2 in API mode cannot be built without R in the PATH or R_HOME defined. Correct this or force ABI mode-only by defining the environment variable RPY2_CFFI_MODE=ABI
    cffi mode: CFFI_MODE.ANY
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/c1/f6/3425652de9f0479f31fd87e62ac08cd8c765af53aff59db915878b55f1b7/rpy2-3.3.4.tar.gz#sha256=763601b1c5d0195a53341c5b053b419e48eab163cdb8908e313ce7c1f207a3d1 (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.3.3.tar.gz (173 kB)
     |████████████████████████████████| 173 kB 28.2 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_53aa1c8b4842472591cc0c7c464acfdb/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_53aa1c8b4842472591cc0c7c464acfdb/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-5z93_bqt
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_53aa1c8b4842472591cc0c7c464acfdb/
    Complete output (2 lines):
    Error: rpy2 in API mode cannot be built without R in the PATH or R_HOME defined. Correct this or force ABI mode-only by defining the environment variable RPY2_CFFI_MODE=ABI
    cffi mode: CFFI_MODE.ANY
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/e2/ec/5f88d44ce8a0c721cd1a78f56717d765e53cf205730d8af1cc3482749060/rpy2-3.3.3.tar.gz#sha256=ca578dae6df117d55e3e547b29a8247f3710b1246c322c6a6cec3249b823cf0b (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.3.2.tar.gz (172 kB)
     |████████████████████████████████| 172 kB 34.5 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_d798dd5b2db74b2c8f31d39e74ce2ce5/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_d798dd5b2db74b2c8f31d39e74ce2ce5/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-n0onm0j2
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_d798dd5b2db74b2c8f31d39e74ce2ce5/
    Complete output (2 lines):
    Error: rpy2 in API mode cannot be built without R in the PATH or R_HOME defined. Correct this or force ABI mode-only by defining the environment variable RPY2_CFFI_MODE=ABI
    cffi mode: CFFI_MODE.ANY
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/58/be/90cbc6b530a86a2d927bb8379983c7b17e0b89ca844c27270fc4509d1dbf/rpy2-3.3.2.tar.gz#sha256=2e12c99e56a1faa68f3b82a71b90e3389203c1e76cd53fe38946a8a7a31f9750 (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.3.1.tar.gz (172 kB)
     |████████████████████████████████| 172 kB 31.6 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_8f1c4f9bf0a1428f9d4e4a24e7a237e6/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_8f1c4f9bf0a1428f9d4e4a24e7a237e6/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-16w3ix3b
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_8f1c4f9bf0a1428f9d4e4a24e7a237e6/
    Complete output (2 lines):
    Error: rpy2 in API mode cannot be built without R in the PATH or R_HOME defined. Correct this or force ABI mode-only by defining the environment variable RPY2_CFFI_MODE=ABI
    cffi mode: CFFI_MODE.ANY
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/72/9b/38a6970066ee42cc3455505db284f5636e6b3bfd3520452beb53b1ea62ac/rpy2-3.3.1.tar.gz#sha256=3f25f45d327e2244f0d0654a2cf11d3e8a402ecf73cab5662e37610ab47bbee8 (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.3.0.tar.gz (172 kB)
     |████████████████████████████████| 172 kB 31.6 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_91a8a8b2d5504db8804cea9716f78d8d/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_91a8a8b2d5504db8804cea9716f78d8d/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-2jhf8jql
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_91a8a8b2d5504db8804cea9716f78d8d/
    Complete output (2 lines):
    Error: rpy2 in API mode cannot be built without R in the PATH or R_HOME defined. Correct this or force ABI mode-only by defining the environment variable RPY2_CFFI_MODE=ABI
    cffi mode: CFFI_MODE.ANY
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/f7/ec/07609b4f26c9dc800374dc00b64ce94ab2ec5792d9ce50d46529dfdf5bc1/rpy2-3.3.0.tar.gz#sha256=08c90377dce51126311a5fa6e1cd7d1f451f759ce1f5e1755d8f44e8c40c1f23 (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.2.7.tar.gz (162 kB)
     |████████████████████████████████| 162 kB 31.0 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_f67d98c01b8a413d93c640ef6c6e3098/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_f67d98c01b8a413d93c640ef6c6e3098/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-8jhi3aku
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_f67d98c01b8a413d93c640ef6c6e3098/
    Complete output (1 lines):
    There is no R_HOME and the R executable cannot be found.
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/39/c0/61120f9dae06b4887426d229b68a7a5f0ca1f9cb3986319bb9484819a28d/rpy2-3.2.7.tar.gz#sha256=db9c71eec3c91f44373839c24a1888a3999f2f342300e8e696e5dcc047d3df6d (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.2.6.tar.gz (162 kB)
     |████████████████████████████████| 162 kB 33.4 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_54b80bc48a4b468286af84340b94c095/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_54b80bc48a4b468286af84340b94c095/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-zv9vc0pu
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_54b80bc48a4b468286af84340b94c095/
    Complete output (1 lines):
    There is no R_HOME and the R executable cannot be found.
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/2f/9a/bb5444499e9cefbe2696a56bd4368b163d72ddc6c97b1e940a3579882e07/rpy2-3.2.6.tar.gz#sha256=1829c23fd9f6ac9dc8cf0ea14c1bc20a2a6e4a35c6f2c4726857dcd1310329dd (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.2.5.tar.gz (162 kB)
     |████████████████████████████████| 162 kB 26.2 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_5c892293f0c74751ad5747fff08d2bd2/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_5c892293f0c74751ad5747fff08d2bd2/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-si_waw1d
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_5c892293f0c74751ad5747fff08d2bd2/
    Complete output (1 lines):
    There is no R_HOME and the R executable cannot be found.
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/bb/6b/074f69f3247254d44b6d98218739fc826886480c9fe160d2dcff94e325f0/rpy2-3.2.5.tar.gz#sha256=33fbb7c244fe4151a47899ab8f2f138ca0042d25755a3956d584453a8719d35e (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.2.4.tar.gz (162 kB)
     |████████████████████████████████| 162 kB 33.8 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_44b30be1f8c946db956834ddd0099143/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_44b30be1f8c946db956834ddd0099143/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-67kz8vtk
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_44b30be1f8c946db956834ddd0099143/
    Complete output (1 lines):
    There is no R_HOME and the R executable cannot be found.
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/62/bc/d43df0d9e96a38985a97d1cbdb5722e10cd8fa0da45686972f0b8fd18a67/rpy2-3.2.4.tar.gz#sha256=3daf1a4b28c4e354ef989093f03b066908bf6e5082a6f4af72cc3fd928a28dc6 (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.2.3.tar.gz (162 kB)
     |████████████████████████████████| 162 kB 30.6 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_137ccb6e27a24975aa960366d2c97694/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_137ccb6e27a24975aa960366d2c97694/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-son6xper
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_137ccb6e27a24975aa960366d2c97694/
    Complete output (1 lines):
    There is no R_HOME and the R executable cannot be found.
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/8e/bf/13faf3607f08bed30646e81d21ff4005b99f3963f94979c09113c09a0046/rpy2-3.2.3.tar.gz#sha256=3cd58cffaa580f3a0053dc7e8be61e810e8b046c43a8948586010572744d0391 (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.2.2.tar.gz (162 kB)
     |████████████████████████████████| 162 kB 32.6 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_08466ef8acc640e7968dd7da0e1411cc/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_08466ef8acc640e7968dd7da0e1411cc/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-q6hqreql
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_08466ef8acc640e7968dd7da0e1411cc/
    Complete output (1 lines):
    There is no R_HOME and the R executable cannot be found.
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/d9/51/d5c41ac70e0aa3272f24cf836bd35d6246c95b98d8d811f338cec5a8098e/rpy2-3.2.2.tar.gz#sha256=b71757776f1c652570f07243e708b6e6626456d3ee7d1f336357d4d393bd722c (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.2.1.tar.gz (161 kB)
     |████████████████████████████████| 161 kB 26.3 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_b247fdd0efc54edeaf58de35eef3e79b/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_b247fdd0efc54edeaf58de35eef3e79b/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-yrcebmmn
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_b247fdd0efc54edeaf58de35eef3e79b/
    Complete output (15 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_b247fdd0efc54edeaf58de35eef3e79b/setup.py", line 102, in <module>
        c_extension_status = get_r_c_extension_status()
      File "/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_b247fdd0efc54edeaf58de35eef3e79b/setup.py", line 91, in get_r_c_extension_status
        *situation.get_r_flags(r_home, '--ldflags')
      File "/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_b247fdd0efc54edeaf58de35eef3e79b/rpy2/situation.py", line 175, in get_r_flags
        _get_r_cmd_config(r_home, flags,
      File "/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_b247fdd0efc54edeaf58de35eef3e79b/rpy2/situation.py", line 141, in _get_r_cmd_config
        r_exec = get_r_exec(r_home)
      File "/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_b247fdd0efc54edeaf58de35eef3e79b/rpy2/situation.py", line 130, in get_r_exec
        r_exec = os.path.join(r_home, 'bin', 'R')
      File "/Users/Angela/opt/anaconda3/lib/python3.9/posixpath.py", line 76, in join
        a = os.fspath(a)
    TypeError: expected str, bytes or os.PathLike object, not NoneType
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/27/1e/864f9d5c685a913e5f6ee8f9ff4779db9826964740b1b92a5cfe74d5094c/rpy2-3.2.1.tar.gz#sha256=cfba1b5d5ad56f9f52a0df4df8015a342043f113576003967fd760e063023ca6 (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.2.0.tar.gz (161 kB)
     |████████████████████████████████| 161 kB 27.1 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Angela/opt/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_8c3cd7c823f741a7be6eee22f480b3fe/setup.py'"'"'; __file__='"'"'/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_8c3cd7c823f741a7be6eee22f480b3fe/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-pip-egg-info-els69bia
         cwd: /private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_8c3cd7c823f741a7be6eee22f480b3fe/
    Complete output (15 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_8c3cd7c823f741a7be6eee22f480b3fe/setup.py", line 102, in <module>
        c_extension_status = get_r_c_extension_status()
      File "/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_8c3cd7c823f741a7be6eee22f480b3fe/setup.py", line 91, in get_r_c_extension_status
        *situation.get_r_flags(r_home, '--ldflags')
      File "/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_8c3cd7c823f741a7be6eee22f480b3fe/rpy2/situation.py", line 175, in get_r_flags
        _get_r_cmd_config(r_home, flags,
      File "/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_8c3cd7c823f741a7be6eee22f480b3fe/rpy2/situation.py", line 141, in _get_r_cmd_config
        r_exec = get_r_exec(r_home)
      File "/private/var/folders/qw/j72d9nwn5dq0vxtqtw2bbtnh0000gp/T/pip-install-0syq85au/rpy2_8c3cd7c823f741a7be6eee22f480b3fe/rpy2/situation.py", line 130, in get_r_exec
        r_exec = os.path.join(r_home, 'bin', 'R')
      File "/Users/Angela/opt/anaconda3/lib/python3.9/posixpath.py", line 76, in join
        a = os.fspath(a)
    TypeError: expected str, bytes or os.PathLike object, not NoneType
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/7e/e0/7da849bb6cf47466ceb28a75f930e61c311878882c275dfb4bbb4fdcc3cb/rpy2-3.2.0.tar.gz#sha256=18025f53c1a32b7b38ed319c81f8445b9a8809e0ecba210634b67764d68da978 (from https://pypi.org/simple/rpy2/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading rpy2-3.1.0.tar.gz (158 kB)
     |████████████████████████████████| 158 kB 26.4 MB/s 
Requirement already satisfied: tenacity>=6.2.0 in ./opt/anaconda3/lib/python3.9/site-packages (from plotly) (8.0.1)
Requirement already satisfied: six in ./opt/anaconda3/lib/python3.9/site-packages (from plotly) (1.16.0)
Requirement already satisfied: tqdm in ./opt/anaconda3/lib/python3.9/site-packages (from finance-datareader) (4.62.3)
Requirement already satisfied: lxml in ./opt/anaconda3/lib/python3.9/site-packages (from finance-datareader) (4.6.3)
Requirement already satisfied: requests-file in ./opt/anaconda3/lib/python3.9/site-packages (from finance-datareader) (1.5.1)
Requirement already satisfied: pandas>=0.19.2 in ./opt/anaconda3/lib/python3.9/site-packages (from finance-datareader) (1.3.4)
Requirement already satisfied: requests>=2.3.0 in ./opt/anaconda3/lib/python3.9/site-packages (from finance-datareader) (2.26.0)
Requirement already satisfied: pytest in ./opt/anaconda3/lib/python3.9/site-packages (from rpy2) (6.2.4)
Requirement already satisfied: jinja2 in ./opt/anaconda3/lib/python3.9/site-packages (from rpy2) (2.11.3)
Requirement already satisfied: pytz in ./opt/anaconda3/lib/python3.9/site-packages (from rpy2) (2021.3)
Requirement already satisfied: simplegeneric in ./opt/anaconda3/lib/python3.9/site-packages (from rpy2) (0.8.1)
Collecting tzlocal
  Downloading tzlocal-4.1-py3-none-any.whl (19 kB)
Requirement already satisfied: cffi>=1.0.0 in ./opt/anaconda3/lib/python3.9/site-packages (from rpy2) (1.14.6)
Requirement already satisfied: pycparser in ./opt/anaconda3/lib/python3.9/site-packages (from cffi>=1.0.0->rpy2) (2.20)
Requirement already satisfied: python-dateutil>=2.7.3 in ./opt/anaconda3/lib/python3.9/site-packages (from pandas>=0.19.2->finance-datareader) (2.8.2)
Requirement already satisfied: numpy>=1.17.3 in ./opt/anaconda3/lib/python3.9/site-packages (from pandas>=0.19.2->finance-datareader) (1.20.3)
Requirement already satisfied: charset-normalizer~=2.0.0 in ./opt/anaconda3/lib/python3.9/site-packages (from requests>=2.3.0->finance-datareader) (2.0.4)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in ./opt/anaconda3/lib/python3.9/site-packages (from requests>=2.3.0->finance-datareader) (1.26.7)
Requirement already satisfied: idna<4,>=2.5 in ./opt/anaconda3/lib/python3.9/site-packages (from requests>=2.3.0->finance-datareader) (3.2)
Requirement already satisfied: certifi>=2017.4.17 in ./opt/anaconda3/lib/python3.9/site-packages (from requests>=2.3.0->finance-datareader) (2021.10.8)
Requirement already satisfied: MarkupSafe>=0.23 in ./opt/anaconda3/lib/python3.9/site-packages (from jinja2->rpy2) (1.1.1)
Requirement already satisfied: attrs>=19.2.0 in ./opt/anaconda3/lib/python3.9/site-packages (from pytest->rpy2) (21.2.0)
Requirement already satisfied: iniconfig in ./opt/anaconda3/lib/python3.9/site-packages (from pytest->rpy2) (1.1.1)
Requirement already satisfied: packaging in ./opt/anaconda3/lib/python3.9/site-packages (from pytest->rpy2) (21.0)
Requirement already satisfied: pluggy<1.0.0a1,>=0.12 in ./opt/anaconda3/lib/python3.9/site-packages (from pytest->rpy2) (0.13.1)
Requirement already satisfied: py>=1.8.2 in ./opt/anaconda3/lib/python3.9/site-packages (from pytest->rpy2) (1.10.0)
Requirement already satisfied: toml in ./opt/anaconda3/lib/python3.9/site-packages (from pytest->rpy2) (0.10.2)
Requirement already satisfied: pyparsing>=2.0.2 in ./opt/anaconda3/lib/python3.9/site-packages (from packaging->pytest->rpy2) (3.0.4)
Collecting pytz-deprecation-shim
  Downloading pytz_deprecation_shim-0.1.0.post0-py2.py3-none-any.whl (15 kB)
Collecting tzdata
  Downloading tzdata-2021.5-py2.py3-none-any.whl (339 kB)
     |████████████████████████████████| 339 kB 30.1 MB/s 
Building wheels for collected packages: rpy2
  Building wheel for rpy2 (setup.py) ... done
  Created wheel for rpy2: filename=rpy2-3.1.0-py3-none-any.whl size=169548 sha256=c9e05f65cfc470143d2220242c4d3905b4b92c47e14c22a2a89f38ec1d918142
  Stored in directory: /Users/Angela/Library/Caches/pip/wheels/48/9f/27/be43afa016c73d5e8628b599fc3ea50932c140bfe59b379045
Successfully built rpy2
Installing collected packages: tzdata, pytz-deprecation-shim, tzlocal, rpy2, plotly
  Attempting uninstall: plotly
    Found existing installation: plotly 5.4.0
    Uninstalling plotly-5.4.0:
      Successfully uninstalled plotly-5.4.0
Successfully installed plotly-5.5.0 pytz-deprecation-shim-0.1.0.post0 rpy2-3.1.0 tzdata-2021.5 tzlocal-4.1
(base) Angela@gwagsalaui-MacBook-Pro ~ %  
